src/graph.cc

```diff
@@ -28,11 +28,63 @@
 #include "state.h"
 #include "util.h"
 
+#ifdef HAVE_LIBURING
+#include "uring.h"
+#endif
+
 bool Node::Stat(DiskInterface* disk_interface, string* err) {
   return (mtime_ = disk_interface->Stat(path_, err)) != -1;
 }
 
+#ifdef HAVE_LIBURING
+
+void Node::BulkStatCallback(TimeStamp t, const char *, void *data) {
+  Node *node = (Node *)data;
```

Comment:
Please use `reinterpret_cast`, maybe even like this:
```cpp
reinterpret_cast<Node&>(*data).mtime_ = t;
```

Reply:
Until now, I thought `reinterpret_cast` was new in C++11 so it can't be used in Ninja, but I looked it up and I was wrong. Some aspects of that operator are new in C++11, but the operator is older.